lib/IRGen/IRGenSIL.cpp

```diff
@@ -719,6 +719,28 @@ class IRGenSILFunction :
     }
   }
 
+  // To make it unambiguous whether a `var` binding has been initialized,
+  // zero-initialize the first pointer-sized field. LLDB uses this to
+  // recognize to detect uninitizialized variables. This can be removed once
+  // swiftc switches to @llvm.dbg.addr() intrinsics.
+  void zeroInit(llvm::AllocaInst *AI) {
```

Comment:
it looks like we could just make this an `llvm::AllocaInst &` ?

Reply:
No, the other user doesn't seem to always pass an alloca, so we can't just blatantly make the `dyn_cast<>` a `cast<>` [I tried]. If you think it's a bug, the assertion fires while building the swift standard library.

Reply:
I would like to know: what is it when it isn't an AllocaInst? Just wondering if there is another case we should handle.